lib/Runtime/PlatformAgnostic/Platform/Linux/DateTime.cpp

```diff
@@ -0,0 +1,166 @@
+        // based on the date and time zone we are in.
+        return GetStandardName(nameLength);
+    }
+
+    void Utility::UpdateTimeZoneInfo()
+    {
+        uint32 tickCount = GetTickCount();
+        if (tickCount - lastTimeZoneUpdateTickCount > updatePeriod)
+        {
+            time_t gtime = time(NULL);
+            struct tm ltime;
+
+            tzset();
+            localtime_r(&gtime, &ltime);
+
+            standardZoneNameLength = strlen(ltime.tm_zone);
+
+        #if defined(WCHAR_IS_CHAR16_T)
+            for(int i = 0; i < standardZoneNameLength; i++) {
+                standardZoneName[i] = (char16_t)ltime.tm_zone[i];
+            }
+        #elif defined(WCHAR_IS_WCHAR_T)
+            mbstowcs( (wchar_t*) standardZoneName, ltime.tm_zone,
+                      sizeof(standardZoneName) / sizeof(WCHAR));
+        #else
+        #error "WCHAR should be either wchar_t or char16_t"
+        #endif
+
+            standardZoneName[standardZoneNameLength] = (WCHAR)0;
+            lastTimeZoneUpdateTickCount = tickCount;
+        }
+    }
+
+    // DateTimeHelper ******
+    #define TM_TO_SYSTIME(SYS, TM)           \
+            SYS->wYear = TM->tm_year + 1900; \
+            SYS->wMonth = TM->tm_mon + 1;    \
+            SYS->wDayOfWeek = TM->tm_wday;   \
+            SYS->wDay = TM->tm_mday;         \
+            SYS->wHour = TM->tm_hour;        \
+            SYS->wMinute = TM->tm_min;       \
+            SYS->wSecond = TM->tm_sec
+
+    #define SYSTIME_TO_TM(SYS, TM)           \
+            TM->tm_year = SYS->wYear - 1900; \
+            TM->tm_mon = SYS->wMonth - 1;    \
+            TM->tm_wday = SYS->wDayOfWeek;   \
+            TM->tm_mday = SYS->wDay;         \
+            TM->tm_hour = SYS->wHour;        \
+            TM->tm_min = SYS->wMinute;       \
+            TM->tm_sec = SYS->wSecond
+
+    const bool SysLocalToUtc(SYSTEMTIME *local, SYSTEMTIME *utc)
+    {
+        struct tm local_tm;
+        SYSTIME_TO_TM(local, (&local_tm));
+
+        // tm doesn't have milliseconds
+        // uint32_t msec = utc->wMilliseconds;
+        // simply keep it as is
+
+        tzset();
+        time_t utime = timegm(&local_tm);
+        struct tm local_bck;
+        memcpy(&local_bck, &local_tm, sizeof(struct tm));
+        mktime(&local_bck);
+        utime -= local_bck.tm_gmtoff; // reverse UTC
+
+        struct tm utc_tm = *gmtime (&utime);
+
+        TM_TO_SYSTIME(utc, (&utc_tm));
+        // put milliseconds back
+        utc->wMilliseconds = local->wMilliseconds;
+        return true;
+    }
+
+    const bool SysUtcToLocal(SYSTEMTIME *utc, SYSTEMTIME *local, int64_t &bias, int &offset, bool &isDaylightSavings)
+    {
+        struct tm utc_tm;
+        SYSTIME_TO_TM(utc, (&utc_tm));
+
+        // tm doesn't have milliseconds
+        // uint32_t msec = utc->wMilliseconds;
+        // simply keep it as is
+        tzset();
+        time_t ltime = mktime(&utc_tm);
```

Comment:
Generally, while the code probably works, it's a little weird to read this code- mktime's documentation says that it takes in the local time but here you pass in a tm representing time in UTC- curious why you don't just do the following:

// Set time to represet UTC (which doesn't get affected by Daylight Savings)
utc_tm.tm_isdst = 0;
utc_tm.tm_gmtoff = 0;

time_t utcTime = timegm(&utc_tm);
struct tm local_tm* = localtime(&utcTime);